packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -97,3 +97,140 @@ export const useStaticVirtualizerMeasure = (
     scrollRef,
   };
 };
+
+/**
+ * React hook that measures virtualized space dynamically to ensure optimized virtualization length.
+ */
+export const useDynamicVirtualizerMeasure = (
+  virtualizerProps: VirtualizerMeasureDynamicProps,
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+  scrollRef: (instance: HTMLElement | HTMLDivElement | null) => void;
+} => {
+  const { defaultItemSize, direction = 'vertical', numItems, getItemSize, currentIndex } = virtualizerProps;
+
+  const sizeTracker = React.useRef<number[]>(new Array(numItems).fill(defaultItemSize));
```

Comment:
Maybe I can't read but, I see that he the size tracker is updated in multiple places but never actually read, is it being used for something?

Reply:
Removed - old code from having local/controlled state we now just pass in.